supervisor/store/__init__.py

```diff
@@ -60,12 +67,22 @@ async def _add_repository(url):
             if not await repository.load():
                 _LOGGER.error("Can't load from repository %s", url)
                 return
-            self.repositories[url] = repository
 
             # don't add built-in repository to config
             if url not in BUILTIN_REPOSITORIES:
+                # Verify that it is a add-on repository
+                repository_file = Path(repository.git.path, "repository.json")
+                try:
+                    SCHEMA_REPOSITORY_CONFIG(read_json_file(repository_file))
```

Comment:
This does I/O so shouldn't be done in the event loop. Schedule it on the executor thread pool.